gtest_parallel.py

```diff
@@ -301,12 +301,12 @@ def __fetch_test_output(self, test_name, log_file):
     add useful information in this case
     """
     output = ""
-    start_pattern = re.compile(".*\[ *RUN *\].*" + test_name)
-    success_pattern = re.compile(".*\[ *OK *\].*" + test_name)
-    passed_pattern = re.compile("\[ *PASSED *\]")
-    failure_pattern = re.compile("\[ *FAILED *\]")
-    size_pattern = re.compile("\[ *SIZE *\]")
-    status_pattern = re.compile(".*\[(=*|-*)\].*") # brackets with = or - symbols
+    start_pattern = re.compile(".*[ *RUN *].*" + test_name)
+    success_pattern = re.compile(".*[ *OK *].*" + test_name)
```

Comment:
Hold on. I don't get it. Doesn't this regex match a string with an 'O' or a 'K'?

Reply:
Maybe this should be `r".*\[ *OK *\].*"`?

Reply:
See the 3rd and 4th paragraphs of https://docs.python.org/3/library/re.html

Reply:
I think I see your point. Would this be most correct using the `r` literal route?
```
r".*[ *OK *].*"
```
We're not wanting to match the `\` character literally which use of the `r` literal syntax would be including with your example (I think). I was toying with https://regex101.com/r/iDHrSq/1 to try and figure this out. It accepted/matched using the `\` as well which I didn't expect, but that doesn't really match with the doc you pointed out.

Reply:
`[` and `]` are special characters in a regex used to represent a set of characters. e.g. `"[abc]"` will match 'a', 'b', or 'c'. If you want to match a literal `[` character, you need to escape it with a slash `\'. The problem is that the slash has a different meaning in a python string literal. e.g "\n" is a single newline character. My guess is that "\[" still results in a string containing the '\' and '[' characters but now it generates a warning because "\[" is not a valid escape sequence for a string literal. But it does have meaning for a regex.

I think we want either

```
r".*\[ *OK *\].*"
```

or

```
".*\\[ *OK *\\].*"
```